### DISCUSSION TITLE
Vendor specific host implementation

### DISCUSSION THREAD

**Discussion by Docfch** [cherry-embedded/CherryUSB#179] [Q&A]
Hello,

I have three interconnected hubs and on each hub I have connected an STM32 whose USB is in vendor-specific device mode. I want to plug it to a pico w to establish wifi communication through usb.

Is there an example on vendor specific host implementation ?

Thanks.

**Comment by sakumisu** [comment]
Refer to https://github.com/cherry-embedded/CherryUSB/tree/master/class/vendor

**Reply by Docfch** [reply]
I checked the serial folder but actually i don't have any ftdi or the other chips on the board (raspberry pico w).

**Reply by sakumisu** [reply]
I dont know what is pico w, these url just tells you how to implment a vendor class driver, if you use pico w as a usb host, please give up, i have no plan to support it for usb host driver.

**Reply by Docfch** [reply]
pico w uses rp2040.
and what about esp32 s3 ?

**Reply by sakumisu** [reply]
ESP32S3 uses dwc2 and you can have a try, suggest you to use stm32 as host or other chips with ehci host driver.I have no plan to support rp2040.

**Reply by Docfch** [reply]
unfortunately, my stm32 can work only as a device mode.
